app/src/main/java/org/thoughtcrime/securesms/home/PathActivity.kt

```diff
@@ -240,19 +240,22 @@ class PathActivity : PassphraseRequiredActionBarActivity() {
             dotViewLayoutParams.addRule(CENTER_IN_PARENT)
             dotView.layoutParams = dotViewLayoutParams
             addView(dotView)
-            Handler().postDelayed({
+            postDelayed({
                 performAnimation()
             }, dotAnimationStartDelay)
         }
 
         private fun performAnimation() {
-            expand()
```

Comment:
`performAnimation` is a "recursive" call, basically keep running in the main thread. This "loop" is stored globally so it never gets cleaned up when the view disappears. Here a circuit breaker is introduced so that when the view goes out of windows the animation stops.